prescient/stats/operations_stats.py

```diff
@@ -7,9 +7,250 @@
+
+        self.observed_thermal_dispatch_levels = extractor.get_all_thermal_dispatch_levels(sced)
+        self.observed_thermal_headroom_levels = extractor.get_all_thermal_headroom_levels(sced)
+        self.observed_thermal_states = extractor.get_all_thermal_states(sced)
+        self.observed_costs = extractor.get_cost_per_generator(sced)
+        self.observed_renewables_levels = extractor.get_all_nondispatchable_power_used(sced)
+        self.observed_renewables_curtailment = extractor.get_all_renewables_curtailment(sced)
+
+        self.observed_flow_levels = extractor.get_all_flow_levels(sced)
+
+        self.bus_demands = extractor.get_all_bus_demands(sced)        
+
+        self.observed_bus_mismatches = extractor.get_all_bus_mismatches(sced)
+        self.observed_bus_LMPs = extractor.get_all_bus_LMPs(lmp_sced)
+
+        self.storage_input_dispatch_levels = extractor.get_all_storage_input_dispatch_levels(sced)
+        self.storage_output_dispatch_levels = extractor.get_all_storage_output_dispatch_levels(sced)
+        self.storage_soc_dispatch_levels = extractor.get_all_storage_soc_dispatch_levels(sced)
+
+        self.reserve_requirement = extractor.get_reserve_requirement(sced)
+
+        self.reserve_RT_price = extractor.get_reserve_RT_price(lmp_sced)
+
+    def populate_market_settlement(self,
+                                   sced: OperationsModel,
+                                   extractor: ScedDataExtractor,
+                                   ruc_market: RucMarket,
+                                   time_index: int):
+
+        self.planning_reserve_price = ruc_market.day_ahead_reserve_prices[time_index]
+        self.planning_energy_prices = { b : ruc_market.day_ahead_prices[b,time_index] \
+                                        for b in extractor.get_buses(sced) }
+
+        self.thermal_gen_cleared_DA = { g : ruc_market.thermal_gen_cleared_DA[g,time_index] \
+                                        for g in extractor.get_thermal_generators(sced) }
+
+        self.renewable_gen_cleared_DA = { g : ruc_market.renewable_gen_cleared_DA[g,time_index] \
+                                          for g in extractor.get_nondispatchable_generators(sced) }
+
+        self.thermal_reserve_cleared_DA = { g : ruc_market.thermal_reserve_cleared_DA[g,time_index] \
+                                            for g in extractor.get_thermal_generators(sced) }
+
+        self.thermal_gen_revenue = dict()
+        self.renewable_gen_revenue = dict()
+        for b in extractor.get_buses(sced):
+            price_DA = self.planning_energy_prices[b]
+            price_RT = self.observed_bus_LMPs[b]
+
+            for g in extractor.get_thermal_generators_at_bus(sced, b):
+                self.thermal_gen_revenue[g] = \
+                    self.thermal_gen_cleared_DA[g]*price_DA + \
+                    (self.observed_thermal_dispatch_levels[g] - self.thermal_gen_cleared_DA[g])*price_RT
```

Comment:
For this calculation to be correct, we need to be sure what data EGRET is returning (EGRET should probably always return in $/MWh, but I don't think it does). In general, we also need to offset by the time step duration for non-hourly time-steps.

Reply:
Need to resolve https://github.com/grid-parity-exchange/Egret/issues/177.

Reply:
Once https://github.com/grid-parity-exchange/Egret/pull/178 is merged, the appropriate thing to do will be to multiply this quantity by `sced_period_length_minutes/60.`, and do so similarly for the other revenues reported below.